src/prognostic_equations/zero_velocity.jl

```diff
@@ -1,20 +1,49 @@
+import ClimaCore: MatrixFields
+import LinearAlgebra: UniformScaling
+
 ###
 ### edmfx advection test 
 ###
 
+# Turn off all momentum tendencies in the advection test.
 function zero_velocity_tendency!(Yₜ, Y, p, t)
-    # turn off all monmentum tendencies in the advection test
-    if p.atmos.advection_test
-        FT = eltype(Y)
-        n = n_mass_flux_subdomains(p.atmos.turbconv_model)
+    p.atmos.advection_test || return nothing
+    @. Yₜ.c.uₕ = C12(0, 0)
+    @. Yₜ.f.u₃ = C3(0)
+    if p.atmos.turbconv_model isa PrognosticEDMFX
+        for j in 1:n_mass_flux_subdomains(p.atmos.turbconv_model)
+            @. Yₜ.f.sgsʲs.:($$j).u₃ = C3(0)
+        end
+    end
+    return nothing
+end
 
-        @. Yₜ.c.uₕ = C12(FT(0), FT(0))
-        @. Yₜ.f.u₃ = Geometry.Covariant3Vector(FT(0))
+# Turn off all momentum tendency derivatives in the advection test.
+function zero_velocity_jacobian!(∂Yₜ_err_∂Y, Y, p, t)
+    p.atmos.advection_test || return nothing
+    for ((row_name, col_name), matrix_entry) in pairs(∂Yₜ_err_∂Y)
+        matrix_entry isa Fields.Field || continue
+        if row_name in (MatrixFields.@name(c.uₕ), MatrixFields.@name(f.u₃))
+            set_identity_matrix_entry!(matrix_entry, row_name, col_name)
+        end
         if p.atmos.turbconv_model isa PrognosticEDMFX
-            for j in 1:n
-                @. Yₜ.f.sgsʲs.:($$j).u₃ = Geometry.Covariant3Vector(FT(0))
+            for j in 1:n_mass_flux_subdomains(p.atmos.turbconv_model)
+                if row_name == MatrixFields.FieldName(:f, :sgsʲs, j, :u₃)
+                    set_identity_matrix_entry!(matrix_entry, row_name, col_name)
+                end
             end
         end
     end
-    return nothing
+end
+
+function set_identity_matrix_entry!(matrix_entry, row_name, col_name)
+    identity_matrix_entry_value = if row_name == col_name
+        # TODO: Add a method for one(::Axis2Tensor) to simplify this.
```

Comment:
Is the suggestion to define `one(::Type{T}) where {T <: Axis2Tensor}`? or `one(::T) where {T <: Axis2Tensor}`?

Reply:
I thought we had it. Ah, it looks like we currently only support `AxisVector`s:
```julia
Base.one(::T) where {T <: Geometry.AxisTensor} = one(T)
Base.one(::Type{T}) where {T′, A, S, T <: Geometry.AxisTensor{T′, 1, A, S}} =
    T(axes(T), S(one(T′)))
```

Reply:
Following other definitions of `zero` and `one` in ClimaCore, I would probably define this as
```
one(::T) where {T <: Axis2Tensor} = one(T)
one(::Type{T}) where {T <: Axis2Tensor} = ...
```
I've already defined `one(::BandMatrixRow)`, so adding this will simplify the case for `row_name == col_name` to `-one(eltype(matrix_entry))`

Reply:
It would also probably be good to generalize the method of `one` for `AxisVector`s to
```
one(::Type{T}) where {A, S, T <: Geometry.AxisTensor{<:Any, 1, A, S} = T(A, ones(S))
```
The version you gave above only works for single-component vectors.